bindgen/codegen/mod.rs

```diff
@@ -4487,21 +4528,71 @@ pub(crate) fn codegen(
             result.push(dynamic_items_tokens);
         }
 
-        postprocessing::postprocessing(result.items, context.options())
+        if !result.items_to_serialize.is_empty() {
+            utils::serialize_items(&result, context)?;
```

Comment:
I'd probably move the `items_to_serialize.is_empty()` check to `serialize_items`, and call this unconditionally.